x-pack/plugins/apm/common/ml_job_constants.ts

```diff
@@ -26,10 +26,6 @@ export function getMlJobServiceName(jobId: string) {
   return jobId.split('-').slice(0, -2).join('-');
 }
```

Comment:
Is `getMlJobServiceName` still used? I thought we had stopped parsing the job id and was not instead solely relying on the job groups?

Reply:
Similar with `getMlPrefix` and `getMlJobId`: I think we should be able to get rid of them.

**`getMlPrefix`** is only used in https://github.com/elastic/kibana/blob/04b8a4915100f73cd07cf1d645fbe0e6ca3f90d4/x-pack/plugins/apm/public/services/rest/ml.ts#L72
and I don't think we should be splitting ml jobs in different indices. We should be able to remove the prefix entirely.

**`getMlJobId`** should simply be a random id. A job should be retrieved based on groups - not the job id.

Reply:
We are already adding service name and transaction type to the groups so in theory this change is backwards compatible.

https://github.com/elastic/kibana/blob/04b8a4915100f73cd07cf1d645fbe0e6ca3f90d4/x-pack/plugins/apm/public/services/rest/ml.ts#L58-L62

One backwards incompatible change I would like us to make though, is to prefix groups with a property name:
```ts
 const groups = [ 
   `app:apm`, 
   `service.name:${encodeForMlApi(serviceName)}`, 
   `transaction.type:${encodeForMlApi(transactionType)}`, 
 ]; 
```

Reply:
agreed on all these point. I held off since the templated job id is still used in the client-side ml integration, but i can update it to make them job_id agnostic and introduce keyed:group strings in the context of this PR.

Reply:
Let's sync with the rest of the team before doing the breaking changes (like prefixes)